src/main/java/org/tc/controllers/CourseController.java

```diff
@@ -365,4 +372,64 @@ public ModelAndView deletePost(@PathVariable("id") int id) {
             return new ModelAndView(new RedirectView(ACCESS_DENIED_PAGE));
         }
     }
+
+    @RequestMapping(value = "/courses/{id}/start", method = RequestMethod.GET)
+    public ModelAndView start(@PathVariable("id") int id) {
+        Course course = courseService.getById(id);
+        if (courseService.isOwner(course) && courseService.isReady(course)) {
+            ModelAndView mav = new ModelAndView(START_COURSE_VIEW_NAME);
+            mav.addObject(SINGLE_COURSE_OBJECT_NAME, courseDetailsDTOConverter.convert(course));
+            mav.addObject(HEADER_TITLE, "Start course");
+            return mav;
+        } else {
+            return new ModelAndView(new RedirectView(ACCESS_DENIED_PAGE));
```

Comment:
Sometimes it can be created PermissionChecker that will check permission and throws runtime AccessDeiedException that extends from AppSecuityException

Then you can configure Exception resolver to redirect to page. Such approach can help in removing duplication of redirecting to access denied page.
I do not say that you should rewrite - just alternative approach. 

Another approach can be to verify access in Filter and add filter mapping for guarded rest endpoints.